### ISSUE TITLE
Exception on 1.6.0 when navigating to any DAG

### ISSUE THREAD

**Issue by r39132** [apache/airflow#632]
`````` sqlalchemy.exc.ProgrammingError
ProgrammingError: (psycopg2.ProgrammingError) column "extra" of relation "log" does not exist
LINE 1: ...m, dag_id, task_id, event, execution_date, owner, extra) VAL...
                                                             ^
 [SQL: 'INSERT INTO log (dttm, dag_id, task_id, event, execution_date, owner, extra) VALUES (%(dttm)s, %(dag_id)s, %(task_id)s, %(event)s, %(execution_date)s, %(owner)s, %(extra)s) RETURNING log.id'] [parameters: {'task_id': None, 'extra': "[('dag_id', u'feeback_report')]", 'execution_date': None, 'dttm': datetime.datetime(2015, 11, 14, 1, 56, 34, 45259), 'owner': 'anonymous', 'event': 'graph', 'dag_id': None}]

Traceback (most recent call last)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1836, in __call__
return self.wsgi_app(environ, start_response)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1820, in wsgi_app
response = self.make_response(self.handle_exception(e))
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1403, in handle_exception
reraise(exc_type, exc_value, tb)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1817, in wsgi_app
response = self.full_dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1477, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1381, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
rv = self.dispatch_request()
File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
return self.view_functions[rule.endpoint](**req.view_args)
File "/usr/local/lib/python2.7/dist-packages/Flask_Admin-1.2.0-py2.7.egg/flask_admin/base.py", line 68, in inner
return self._run_view(f, *args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/Flask_Admin-1.2.0-py2.7.egg/flask_admin/base.py", line 359, in _run_view
return fn(self, *args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/Flask_Login-0.2.11-py2.7.egg/flask_login.py", line 755, in decorated_view
return func(*args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/airflow-1.6.0-py2.7.egg/airflow/www/utils.py", line 139, in view_func
return f(*args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/airflow-1.6.0-py2.7.egg/airflow/www/utils.py", line 92, in wrapper
session.commit()
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/session.py", line 801, in commit
self.transaction.commit()
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/session.py", line 392, in commit
self._prepare_impl()
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/session.py", line 372, in _prepare_impl
self.session.flush()
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/session.py", line 2015, in flush
self._flush(objects)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/session.py", line 2133, in _flush
transaction.rollback(_capture_exception=True)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
compat.reraise(exc_type, exc_value, exc_tb)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/session.py", line 2097, in _flush
flush_context.execute()
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/unitofwork.py", line 373, in execute
rec.execute(self)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/unitofwork.py", line 532, in execute
uow
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/persistence.py", line 174, in save_obj
mapper, table, insert)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/orm/persistence.py", line 785, in _emit_insert_statements
execute(statement, params)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 914, in execute
return meth(self, multiparams, params)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/elements.py", line 323, in _execute_on_connection
return connection._execute_clauseelement(self, multiparams, params)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1010, in _execute_clauseelement
compiled_sql, distilled_params
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1146, in _execute_context
context)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1341, in _handle_dbapi_exception
exc_info
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
reraise(type(exception), exception, tb=exc_tb)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/base.py", line 1139, in _execute_context
context)
File "/usr/local/lib/python2.7/dist-packages/sqlalchemy/engine/default.py", line 450, in do_execute
cursor.execute(statement, parameters)
ProgrammingError: (psycopg2.ProgrammingError) column "extra" of relation "log" does not exist LINE 1: ...m, dag_id, task_id, event, execution_date, owner, extra) VAL... ^ [SQL: 'INSERT INTO log (dttm, dag_id, task_id, event, execution_date, owner, extra) VALUES (%(dttm)s, %(dag_id)s, %(task_id)s, %(event)s, %(execution_date)s, %(owner)s, %(extra)s) RETURNING log.id'] [parameters: {'task_id': None, 'extra': "[('dag_id', u'feeback_report')]", 'execution_date': None, 'dttm': datetime.datetime(2015, 11, 14, 1, 56, 34, 45259), 'owner': 'anonymous', 'event': 'graph', 'dag_id': None}]```
``````

**Comment by r39132** [comment]
@mistercrunch See 1.6.0 exception above. I get this when navigating to any DAG. This is a blocker for us.

**Comment by artwr** [comment]
@r39132 , I think you might need to `airflow upgradedb`. A few models were updated.

**Comment by r39132** [comment]
Yep.. actually did that... I found many hundreds of processes running .. bringing the server back up. More in a bit.

**Comment by r39132** [comment]
These seems like a transient error related to having many tasks running.

**Comment by r39132** [comment]
I'm running hourly. It seems that it is not honoring my start datetime!

<img width="854" alt="screenshot 2015-11-13 18 47 36" src="https://cloud.githubusercontent.com/assets/581734/11161632/13a2b1ae-8a37-11e5-9b1e-adb9e220e0c7.png">

This resulting in a lot more load and unwanted processing. 



<img width="1318" alt="screenshot 2015-11-13 18 46 32" src="https://cloud.githubusercontent.com/assets/581734/11161627/e813516a-8a36-11e5-8729-888f5bda329c.png">

**Comment by r39132** [comment]
@mistercrunch @artwr  My airflow app is in a terrible state. Essentially, it is not honoring my start date. It keep trying to execute older tasks. I cannot mark these as success either because I get an exception `Exception: Wait. start_date needs to be before end_date` This happen when I try to stop scheduler execution by marking older runs as successful. Now, the dag is just loading my system with a lot of unwanted work and mostly failing.

<img width="945" alt="screenshot 2015-11-13 18 59 10" src="https://cloud.githubusercontent.com/assets/581734/11161703/cedea7d8-8a38-11e5-9244-0818f6ba83a2.png">

**Comment by r39132** [comment]
And then, I eventually get the error before the system runs out of memory and crashes. 
<img width="1098" alt="screenshot 2015-11-13 19 01 41" src="https://cloud.githubusercontent.com/assets/581734/11161706/0b0718d0-8a39-11e5-9518-722b2385f6af.png">

**Comment by bolkedebruin** [comment]
@r39132 Are you sure the database migrations were successful? The migration code to add the "extra" column does exist and should have been executed. Please verify and otherwise maybe add it manually to your postgres db. Furthermore I would check if you are running 1.6 everywhere. and not mixing multiple versions somehow. Clear out the python packages for older versions.

**Comment by r39132** [comment]
Just reran it :

`$ airflow upgradedb
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.`

**Comment by bolkedebruin** [comment]
can you verify which migration scripts are present? can you confirm 502898887f84_adding_extra_to_log is present? Did you install airflow from pip or git?

Did you try to alter the table by hand?

Btw: your host is externally resolvable to a private address, might not what you want to disclose - username is also disclosed.

**Comment by r39132** [comment]
Hi @bolkedebruin 
I upgraded via git, ran setup.py install, then upgrade db, then restarted my webserver and scheduler. It is possible that the webserver and schedulers were running during the above steps.

It looks like this error is not happening anymore after I killed off a lot of processes and restarted everything.

I confirmed the the migration is present. I have not altered any airflow tables by hand. 

```
./airflow/migrations/versions/502898887f84_adding_extra_to_log.py
./build/lib.linux-x86_64-2.7/airflow/migrations/versions/502898887f84_adding_extra_to_log.py
```

Thanks for the note about the externally resolvable host and username. Rectified, but let me know i there is still something exposed that I missed.

I'm closing this ticket - I have another issue related to hourly start time not being honored, but I will follow up with another ticket.
